contents/docs/plugins/build.md

```diff
@@ -78,9 +79,24 @@ Most options in this file are self-explanatory, but there are a few worth explor
+`config` consists of an array of objects that each pertain to a specific configuration field or markdown explanation for your plugin.
+
+Each object in a config can have the following properties:
+
+|   Key    |                    Type                    |                                                                           Description                                                                           |
+| :------: | :----------------------------------------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   type   | `"string"` or `"attachment"` or `"choice"` | Determines the type of the field - "attachment" asks the user for an upload, and "choice" requires the config object to have a `choices` array, explained below |
+|   key    |                  `string`                  |                                     The key of the plugin config field, used to reference the value from inside the plugin                                      |
+|   name   |                  `string`                  |                                          Displayable name of the field - appears on the plugin setup in the PostHog UI                                          |
+| default  |                  `string`                  |                                                                   Default value of the field                                                                    |
+|   hint   |                  `string`                  |                                             More information about the field, displayed under the in the PostHog UI                                             |
+| markdown |                  `string`                  |                                                             Markdown to be displayed with the field                                                             |
+|  order   |                  `number`                  |                                                                           Deprecated                                                                            |
+| required |                 `boolean`                  |                                               Specifies if the user needs to provide a value for the field or not                                               |
+|  secret  |                 `boolean`                  |                     Secret values are write-only and never shown to the user again - useful for plugins that ask for API Keys, for example                      |
+| choices  |                  `array`                   |                           Only accepted on configs with `type` equal to `"choice"` - an array of choices to be presented to the user                            |
```

Comment:
might be good to say what type the choices must be in... `type = array` ... of what?

Reply:
ah yes of course!